### PR TITLE
Add streaming type for streaming content

### PR DESCRIPTION
**Pull request by lfarrell**
- https://unclibrary.atlassian.net/browse/BXC-4622
- https://unclibrary.atlassian.net/browse/BXC-4623

Relies on https://gitlab.lib.unc.edu/cdr/boxc-ansible/-/merge_requests/272

### CODE REVIEW

**Review by sharonluong — CHANGES_REQUESTED** [review]
I think we should add a test to ValidateContentModelJobTest if there is a Streaming URL but no streaming type.

**Comment by sharonluong** [comment]
Are there accepted values for StreamingType?

File: deposit-app/src/main/java/edu/unc/lib/boxc/deposit/validate/ValidateContentModelJob.java [hunk 1]
```diff
@@ -252,7 +252,8 @@ private void validateMemberOrder(Model model) {
 
     private void validateStreamingProperties(Resource resource) {
         var url = resource.getProperty(Cdr.streamingUrl).getString();
-        if (!url.startsWith(STREAMREAPER_PREFIX_URL)) {
+        var streamingType = resource.hasProperty(Cdr.streamingType);
```
Comment (sharonluong) [inline deposit-app/src/main/java/edu/unc/lib/boxc/deposit/validate/ValidateContentModelJob.java:255]: Are there accepted values for StreamingType?

File: indexing-solr/src/test/java/edu/unc/lib/boxc/indexing/solr/filter/SetStreamingPropertiesFilterTest.java [hunk 2]
```diff
@@ -33,7 +33,7 @@ public void setup() {
         dip = new DocumentIndexingPackage(filePid, null, documentIndexingPackageDataLoader);
         dip.setPid(filePid);
         idb = dip.getDocument();
-        filter = new SetStreamingUrlFilter();
+        filter = new SetStreamingPropertiesFilter();
```
Comment (sharonluong) [inline indexing-solr/src/test/java/edu/unc/lib/boxc/indexing/solr/filter/SetStreamingPropertiesFilterTest.java:36]: should add assertions for streaming type as well

**Review by sharonluong — CHANGES_REQUESTED** [review]
looking pretty good!

**Review by bbpennel — CHANGES_REQUESTED** [review]

File: services-camel-app/src/main/java/edu/unc/lib/boxc/services/camel/streaming/StreamingPropertiesRequestProcessor.java [hunk 3]
```diff
@@ -67,15 +69,18 @@ private String validate(StreamingPropertiesRequest request, PID pid) {
...
             } else {
                 if (!url.startsWith(STREAMREAPER_PREFIX_URL)) {
                     return "URL is not a stream reaper URL";
                 }
             }
-            if (request.getType() == null) {
+            if (type == null) {
                 return "Streaming type is required";
+            } else if (!type.equals(STREAMREAPER_SOUND) && !type.equals(STREAMREAPER_VIDEO)) {
+                return "Streaming type is not a valid stream reaper type";
```
Comment (bbpennel) [inline services-camel-app/src/main/java/edu/unc/lib/boxc/services/camel/streaming/StreamingPropertiesRequestProcessor.java:83]: "Streaming type is not a valid type" so we aren't as tightly coupled with stream reaper.

**Review by bbpennel — CHANGES_REQUESTED** [review]

**Review by sharonluong — APPROVED** [review]

**Pull request merged**